kasa/discover.py

```diff
@@ -35,15 +37,19 @@ def __init__(
         target: str = "255.255.255.255",
         discovery_packets: int = 3,
         interface: Optional[str] = None,
+        authentication: Optional[dict] = None,
```

Comment:
So I'm trying to figure out how to best expose the authentication to the API, passing a dict of user/password feels unclean...

I haven't yet made up my mind, but one possibility would be enclosing the necessary authentication magic inside a separate authentication class? For a downstream user the usage would go something like this:

```
from kasa import SmartPlug, Auth

# usage for devices using the legacy protocol
old_dev = SmartPlug("127.0.0.1")
await old_dev.update()

# using klap with non-empty user/password
auth = Auth(username="foo", password="bar")
dev = SmartPlug("127.0.0.1", auth=auth)

# using klap with an empty user/passwd
dev = SmartPlug("127.0.0.1", auth=Auth())
```
This would allow re-using the same auth easily for multiple instances, as well as making it easy to extend this to new authentication types (e.g., if https gets added, that is hinted by the discovery payload).